docs/src/Combinatorics/partitions.md

```diff
@@ -0,0 +1,36 @@
+# Partitions
+
+```@docs
+    Partition
+```
+
+## Unrestricted partitions
+
+```@docs
+    partitions(::Integer)
+    num_partitions(::Integer)
+    ascending_partitions
+```
+
+## Restricted partitions
+
+```@docs
+    partitions(::Integer, ::Integer, ::Integer, ::Integer)
+    partitions(::Integer, ::Integer)
+    num_partitions(::Integer, ::Integer)
+    partitions(::Array{Integer,1}, ::Integer, ::Array{Integer,1}, ::Integer)
```

Comment:
Please don't use `Array{Integer,1}`, use `Vector{Integer}` instead (they are semantically the same, but the latter is the default name in code and when printing these types since Julia 1.6).

(This is not a blocker for this PR, it could be changed later, but it's good to be aware of this)